_overviews/scala3-book/concurrency.md

```diff
@@ -0,0 +1,271 @@
+---
+title: Concurrency
+type: chapter
+description: This page discusses how Scala concurrency works, with an emphasis on Scala Futures.
+num: 68
+previous-page: ca-summary
+next-page: scala-tools
+---
+
+
+When you want to write parallel and concurrent applications in Scala, you *can* use the native Java `Thread` — but the Scala [Future](https://www.scala-lang.org/api/current/scala/concurrent/Future$.html) makes parallel/concurrent programming much simpler, so it’s preferred, and covered in this chapter.
+
+
+
+## Introduction
+
+Here’s a description of the Scala `Future` from its Scaladoc:
+
+>“A `Future` represents a value which may or may not *currently* be available, but will be available at some point, or an exception if that value could not be made available.” 
+
+To help demonstrate what that means, let’s first look at single-threaded programming. In the single-threaded world you bind the result of a method call to a variable like this:
+
+```scala
+def aShortRunningTask(): Int = 42
+val x = aShortRunningTask()
+```
+
+In this code, the value `42` is immediately bound to the variable `x`.
+
+When you’re working with a `Future`, the assignment process looks similar:
+
+```scala
+def aLongRunningTask(): Future[Int] = ???
+val x = aLongRunningTask()
+```
+
+The main difference in this case is that because `aLongRunningTask` takes an indeterminate amount of time to return, the value in `x` may or may not be *currently* available, but it will be available at some point — in the future.
+
+In this chapter you’ll see how to use futures, including how to run multiple futures in parallel and combine their results in a `for` expression. You’ll also see other methods that are used to handle the value in a future once it returns.
+
```

Comment:
I think it might be useful to reason in blocking terms too:
```scala
def aShortRunningTask(): Int = 42
val x = aShortRunningTask()
println("Here")
```
The print is executed only after the `aShortRunningTask` completes. In the second case, the print can be executed even if the computation is not completed